kolibri/plugins/coach/assets/src/modules/coachNotifications/getters.js

```diff
@@ -1,159 +1,185 @@
+      if (!lessonMatch) {
+        return null;
+      }
+    }
+    const groups = notification.assignment_collections
+      .map(idx => classSummary.learnerGroups[idx])
+      .filter(Boolean);
+
+    let collection;
+    if (!groups.length) {
+      collection = {
+        id: classSummary.classId,
+        name: classSummary.className,
+        type: CollectionTypes.CLASSROOM,
+      };
+    } else {
+      const groupMatch =
+        find(groups, group => {
+          return group.member_ids.includes(notification.user_id);
+          // If the learner was removed from all groups, generate the notification for
+          // the first group
+        }) || groups[0];
+      collection = {
+        id: groupMatch.id,
+        name: groupMatch.name,
+        type: CollectionTypes.LEARNERGROUP,
+      };
+    }
+
+    let assignment = {};
+    if (object === QUIZ) {
+      assignment = {
+        name: classSummary.exams[notification.quiz_id].title,
+        type: ContentNodeKinds.EXAM,
+        id: notification.quiz_id,
+      };
+    } else {
+      assignment = {
+        name: classSummary.lessons[notification.lesson_id].title,
+        type: ContentNodeKinds.LESSON,
+        id: notification.lesson_id,
+      };
+    }
+
+    return {
+      ...notification,
+      object,
+      collection,
+      id: Number(notification.id),
+      assignment,
+      resource: getResource(notification.contentnode_id),
+      learnerSummary: {
+        firstUserName: classSummary.learners[notification.user_id].name,
+        firstUserId: notification.user_id,
+        total: 1,
+      },
+    };
+  }
+
+  return orderBy(
+    state.notifications.map(reshapeNotification).filter(n => n),
+    'timestamp',
+    ['desc']
+  );
+}
+
 export function summarizedNotifications(state, getters, rootState, rootGetters) {
   const summaryEvents = [];
   const classSummary = rootGetters['classSummary/notificationModuleData'];
 
   // Group notifications by certain shared values
   // Resource Completed/Needs Help - Same Node and Lesson
   // Lesson/Quiz Completed - Same Lesson/Quiz
-  const groupedNotifications = groupBy(state.notifications, n => {
+  const groupedNotifications = groupBy(getters.allNotifications, n => {
     if (n.object === RESOURCE) {
       // Contains both Needs Help and Resource Completed-typed notifications
-      return `${n.type}_${n.lesson_id}_${n.contentnode_id}`;
+      return `${n.object}_${n.lesson_id}_${n.contentnode_id}`;
     }
     if (n.object === LESSON) {
-      return `${n.type}_${n.lesson_id}`;
+      return `${n.object}_${n.lesson_id}`;
     }
     if (n.object === QUIZ) {
-      return `${n.type}_${n.quiz_id}`;
+      return `${n.object}_${n.quiz_id}`;
     }
   });
 
   for (let groupCode in groupedNotifications) {
-    const allEvents = groupedNotifications[groupCode];
+    // Filter out all bust the most recent event for each user
+    const allEvents = sortedUniqBy(groupedNotifications[groupCode], 'user_id');
```

Comment:
To fix the issue I've had with missing `NeedsHelp` notifications, I filter `groupedNotifications[groupCode]` with a predicate `({event}) => event !== NotificationEvents.ANSWERED`.

Reply:
Screenshot after filtering out `Answered` events.

<img width="538" alt="CleanShot 2020-07-29 at 14 13 43@2x" src="https://user-images.githubusercontent.com/10248067/88854254-da453880-d1a5-11ea-9e7e-ec4957506c28.png">

Reply:
Ah yep - I am filtering the ANSWERED events, but doing so after this step, so just moving it up would be the answer here. Good catch.

Reply:
This should now be addressed - as long as Travis is fine, we should be good to merge.